articles/151_node_to_participant_mapping.md

```diff
@@ -0,0 +1,155 @@
+
+Identification of `Clients` and `Services` happens according to the ros conventions for their topic names (see [	
+Topic and Service name mapping to DDS](140_topic_and_service_name_mapping.md)).
+
+This topic is considered an implementation detail, and not all `rmw` implementations have to use it.
+Thus, all the necessary logic has to be in the rmw implementation itself or in a downstream package.
+Implementing this logic in `rcl` would make it part of the API, and not an implementation detail.
+
+To avoid code repetition, a common implementation of the logic was done in [rmw_dds_common](https://github.com/ros2/rmw_dds_common/).
+
+#### Details of the ros discovery topic
+
+- topic name: `ros_discovery_info`
+- writer qos:
+  - durability: transient local
+  - history: keep last
+  - history depth: 1
+  - reliability: reliable
+- reader qos:
+  - durability: transient local
+  - history: keep all
+  - history depth: 1
+  - reliability: reliable
+
+## Other implications
+
+#### Security
+
+Previously, each node could have different security artifacts.
+That was possible because each node was mapped to one `Participant`.
+The new approach allows to specify different security artifacts for each process.
+For more details, see [ROS 2 Security Enclaves](ros2_security_enclaves.md).
+
+#### Ignore local publications option
+
+There's an `ignore_local_publications` option that can be set when [creating a subscription](https://github.com/ros2/rmw/blob/2250b3eee645d90f9e9d6c96d71ce3aada9944f3/rmw/include/rmw/rmw.h#L517).
+That option avoids receiving messages from `Publishers` within the same `Node`.
+This wasn't implemented in all the rmw implementations (e.g.: [FastRTPS](https://github.com/ros2/rmw_fastrtps/blob/099f9eed9a0f581447405fbd877c6d3b15f1f26e/rmw_fastrtps_cpp/src/rmw_subscription.cpp#L118)).
+
+After this change, implementing this feature will be less direct.
+Some extra logic needs to be added in order to identify from which `Node` a `Publisher` was created.
+
+## Alternative implementations
+
+### Using keyed topics
+
+Keyed topics could be used, with the participant gid as the key.
+That would allow the reader side of the topic to use a keep last, depth 1 history.
+
+### Using participant/writer/reader user data
+
+Instead of using a custom topic to share the extra ROS specific discovery information, a combination of participant/reader/writer user data could be used.
+e.g.:
+
+- Participant user data: list of nodes created by the participant (updated when node created destroyed).
+- Reader user data: node name/namespace.
+- Writer user data: node name/namespace.
+
+That information would be enough to satisfy ros2 required graph API.
+This mechanism will also avoid to have a topic where all nodes need to have write and read access.
+
+This alternative wasn't implemented because of lack of support in some of the currently supported DDS vendors.
```

Comment:
I remember when I started the implementation of this, some of the user data qos weren't implemented in FastRtps.

It seems that now they are, @richiware can you clarify? If it's supported now, I want to make that clear in the document.

Reply:
Sorry for the delay. I missed the mention.
Yes, FastDDS supports Participant/Reader/Writer UserData QoS.